pkg/controllers/job/job_controller_actions.go

```diff
@@ -510,9 +510,25 @@ func (cc *jobcontroller) isDependsOnPodsReady(task string, job *batch.Job) bool
 	runningPodCount := 0
 	for _, podName := range dependsOnPods {
 		pod, err := cc.podLister.Pods(job.Namespace).Get(podName)
+		// If failed to get pod. There are 2 possibilities.
+		// 1. pod not found
+		// 2. Error communicating with apiserver
 		if err != nil {
-			klog.Errorf("Failed to get pod %v/%v %v", job.Namespace, podName, err)
-			continue
+			// If pod is not found. There are 2 possibilities.
```

Comment:
It's OK for me. Some small tip:
* The commments can be reorganized to be simple.
* The last `else` branch is redudant and can be removed.

Reply:
@Thor-wl 
Thanks for you suggestions. I made some changes.